### PR TITLE
Feature: Scene Transition

### PR DESCRIPTION
**Pull request by AHinojal**
We can use a transition between the scenes after the next scene's loading.

The transitition we can create it with an image with the script `SceneTransition`. We can control in the inspector:
- transitionMove: The direction of the transition. 
To put it in the inspector, I have to do it with a integer which come from an enum.
-- 0: Left to Right
-- 1: Right to Left
-- 2: Top to Bottom
-- 3: Bottom to Top
-- 4: Fade In
-- 5: Fade Out
- speedTransition: The speed of the transition.

Also I control the possibility to change of resolution by the user, resizing and positioning the object.
**_The only thing to improve is if the scene is like the level1 o 2 (in other words, big to load), the feeling the first time to load the scene is that the game brokes. It's only the first time to load the big scene_**

_NOTE: When we mount the scenes, be careful always to add the gameobject which do the transition to the script to its corresponding button. I said that because if don't do the transition and load the next scene, can be for this and it's easy to forget it._

I create a prefab `CanvasTransition.prefab` with two example of transitions:
- ImageStart: Goes a image in black from the left ot the right
- ImageFadeOut_Exit: Does a fade out who ends with a black screen.

I put also the scenes to testing the feature (I already tested in ReleaseEditor):
[Scenes_Transitions.zip](https://github.com/PenteractStudios/Penteract/files/6754456/Scenes_Transitions.zip)

GIFS:
ezgif com-gif-maker (5)
ezgif com-gif-maker (6)
ezgif com-gif-maker (7)

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-kdargf Scene transition - Black fade for example](https://app.clickup.com/t/kdargf)

File: Penteract/Assets/Scripts/SceneTransition.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			if (sceneUID != 0) {
+				isExit ? SceneManager::ExitGame() : SceneManager::ChangeScene(sceneUID);
+				if (Time::GetDeltaTime() == 0.f) Time::ResumeGame();
+			}
+		}
+		else {
+			// Transition
+			// Size should be equal then size of the screen
+			if (transitionMove == (int)TransitionMove::LEFT_TO_RIGHT) {
+				transform2D->SetPosition(float3(transform2D->GetPosition().x + (1.f * speedTransition), 0.f, 0.f));
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/SceneTransition.cpp:43]: If the GameObject with the transition hasn't had a ComponentImage this line will crash. Add a check of nullity at the beginning of the update.

File: Penteract/Assets/Scripts/SceneTransition.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		if (finishedTransition) {
+			if (sceneUID != 0) {
+				isExit ? SceneManager::ExitGame() : SceneManager::ChangeScene(sceneUID);
+				if (Time::GetDeltaTime() == 0.f) Time::ResumeGame();
+			}
+		}
+		else {
+			// Transition
+			// Size should be equal then size of the screen
+			if (transitionMove == (int)TransitionMove::LEFT_TO_RIGHT) {
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/SceneTransition.cpp:42]: The use of static_cast is preferred.

File: Penteract/Assets/Scripts/SceneTransition.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	void StartTransition(bool isExit_ = false); // To iniciate the transition
+	void UpdateObjectToResolution(); // To update the size and the position of the image which made the transition
+
+public:
+
+	UID sceneUID = 0;
+	UID transitionUID = 0;
+
+	int transitionMove = (int) TransitionMove::LEFT_TO_RIGHT;
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/SceneTransition.h:34]: static_cast preferred

**Review by Jordi-Gil — DISMISSED** [review]

File: Penteract/Assets/Scripts/SceneTransition.cpp (hunk 1 above)
Comment (AHinojal) [reply Penteract/Assets/Scripts/SceneTransition.cpp:44]: Fixed checking if the GameObject has a ComponentTransform2D and ComponentImage at the beginning of the update ;)

File: Penteract/Assets/Scripts/SceneTransition.cpp (hunk 2 above)
Comment (AHinojal) [reply Penteract/Assets/Scripts/SceneTransition.cpp]: Fixed in all appearances!

File: Penteract/Assets/Scripts/SceneTransition.h (hunk 3 above)
Comment (AHinojal) [reply Penteract/Assets/Scripts/SceneTransition.h]: Fix it!

**Review by Jordi-Gil — APPROVED** [review]

**Review by devildrake — APPROVED** [review]

**Review by isaacfdez — APPROVED** [review]

**Pull request merged**